### PR TITLE
feature/19713 - submit without pins

### PR DESCRIPTION
**Pull request by pris54**
Stop login submit when school or pupil pin is missing.

### CODE REVIEW

**Comment by GuyHarwood** [comment]
@pris54 Does the job - did you have a chat with the BAs about whether to display any instruction to the user?

**Comment by pris54** [comment]
@jon-shipley @GuyHarwood Could this code please be reviewed again? There is a new AC: submit button should be disabled until basic form validation is OK.

File: pupil-spa/src/app/login/login.component.html [hunk 1]
```diff
@@ -15,18 +15,18 @@ <h1 class="heading-xlarge">Multiplication tables check</h1>
         <p class="lede">
             Enter your school password and pupil PIN, then click <span class="nowrap">‘Sign in’</span>
         </p>
-        <form (ngSubmit)="onSubmit(schoolPin.value, pupilPin.value)" autocomplete="off" autocapitalize="none">
+        <form (ngSubmit)="onSubmit(schoolPin.value, pupilPin.value)" #loginForm="ngForm" autocomplete="off" autocapitalize="none">
             <div class="form-group">
-                <label for="school-pin" class="form-label-bold"> School password </label>
-                <input #schoolPin id="school-pin" class="form-control" spellcheck="false" autocorrect="off">
+                <label for="schoolPin" class="form-label-bold"> School password </label>
+                <input #schoolPin id="schoolPin" class="form-control" required minlength="8" maxlength="8" [(ngModel)]="loginModel.schoolPin" name="schoolPin" spellcheck="false" autocorrect="off">
```
Comment (GuyHarwood) [inline pupil-spa/src/app/login/login.component.html]: i am a bit dubious about putting these exact length ranges on the input field.  It means we have more than one place to manage password requirements, and also makes it very obvious what the requirements are to anyone 'trying' to get in.  🤔 would the `required` attribute still do enough to fulfill the requirement?

Comment (jon-shipley) [reply pupil-spa/src/app/login/login.component.html]: Required would technically work, but as different browsers throw up different messages it fails to meet GDS design criteria.

The usual approach would be to manage the UI using Javascript providing the 'red-box' GDS error message.

https://govuk-elements.herokuapp.com/errors/#example-personal-details

**Pull request merged**